frontend/features/page/sections/ContactsSection/ContactsSection.tsx

```diff
@@ -0,0 +1,753 @@
+                transition="0.3s all"
+                backgroundColor="rgba(59,89,152, 0.7)"
+                _hover={{
+                  backgroundColor: "rgba(59,89,152, 1)",
+                }}
+                pos="relative"
+                w="60px"
+                h="60px"
+              >
+                <FacebookIcon color="white" />
+              </Splash>
+            </Box>
+          </Link>
+
+          <Link href="https://www.github.com/inkOfPixel" passHref>
+            <Box as="a">
+              <Splash
+                mx="1"
+                className="contactBubble"
+                transition="0.3s all"
+                backgroundColor="rgba(24,23,23, 0.7)"
+                _hover={{
+                  backgroundColor: "rgba(24,23,23, 1)",
+                }}
+                pos="relative"
+                w="60px"
+                h="60px"
+              >
+                <GithubIcon color="white" />
+              </Splash>
+            </Box>
+          </Link>
+        </Flex>
+      </Flex>
+    </Box>
+  );
+}
+
+enum FormStatus {
+  Idle = "idle",
+  Submitting = "submitting",
+  Submitted = "submitted",
+}
+
+const blankForm = {
+  status: FormStatus.Idle,
+  values: { name: "", email: "", message: "" },
+  validationErrors: { name: null, email: null, message: null },
+  submitError: null,
+};
+
+interface UpdateFieldAction {
+  type: FormActionType.UpdateField;
+  name: string;
+  value: string;
+}
+
+interface SuccessAction {
+  type: FormActionType.Success;
+}
+
+interface FailAction {
+  type: FormActionType.Failed;
+}
+
+interface SubmitAction {
+  type: FormActionType.Submit;
+}
+
+enum FormActionType {
+  UpdateField = "update-field",
+  Success = "success",
+  Failed = "failed",
+  Submit = "submit",
+  ValidationFailed = "validation - failed",
+}
+
+interface ValidationFailed {
+  type: FormActionType.ValidationFailed;
+  validationErrors: FormState["validationErrors"];
+}
+
+type FormAction =
+  | UpdateFieldAction
+  | SuccessAction
+  | FailAction
+  | SubmitAction
+  | ValidationFailed;
+
+interface FormState {
+  status: FormStatus;
+  values: {
+    name: string;
+    email: string;
+    message: string;
+  };
+  validationErrors: {
+    name: Nullable<string>;
+    email: Nullable<string>;
+    message: Nullable<string>;
+  };
+  submitError: Nullable<string>;
+}
+
+function reducer(state: FormState, action: FormAction) {
+  switch (action.type) {
+    case FormActionType.UpdateField: {
+      return {
+        ...state,
+        values: {
+          ...state.values,
+          [action.name]: action.value,
+        },
+        validationErrors: {
+          ...state.validationErrors,
+          [action.name]: null,
+        },
+        submitError: null,
+      };
+    }
+
+    case FormActionType.Submit: {
+      //Check errors
+
+      const hasNoErrors = Object.values(state.validationErrors).every(
+        (prop) => prop == null || prop === ""
+      );
```

Comment:
No need to do this, you're already checking if there are validation errors on the `onFormSubmit` event handler